MdeModulePkg/Core/Dxe/Misc/MemoryProtection.c

```diff
@@ -881,48 +1180,65 @@ MemoryProtectionExitBootServicesCallback (
 **/
 VOID
 EFIAPI
-DisableNullDetectionAtTheEndOfDxe (
+DisableNullDetectionEventFunction (
   EFI_EVENT  Event,
   VOID       *Context
   )
+{
+  DisableNullDetection ();
+  CoreCloseEvent (Event);
+  return;
+}
+
+// MU_CHANGE [END]
+
+// MU_CHANGE START: Add function to enable null detection as it is now done in DXE instead of PEI
+
+/**
+  Enable NULL pointer detection by changing the attributes of page 0. The assumption is that PEI
+  has set page zero to allocated so this operation can be done safely.
+
+  @retval EFI_SUCCESS       Page zero successfully marked as read protected
+  @retval Other             Page zero could not be marked as read protected
+
+**/
+VOID
+EFIAPI
+EnableNullDetection (
+  VOID
+  )
 {
   EFI_STATUS                       Status;
   EFI_GCD_MEMORY_SPACE_DESCRIPTOR  Desc;
 
-  DEBUG ((DEBUG_INFO, "DisableNullDetectionAtTheEndOfDxe(): start\r\n"));
-  //
-  // Disable NULL pointer detection by enabling first 4K page
-  //
   Status = CoreGetMemorySpaceDescriptor (0, &Desc);

```

Comment:
The original code asserted when failing to get the page 0 descriptor and this code does not. I assume that was an intentional decision?

Reply:
I'll add it back. I think it is good to have that information.

Reply:
This also should goto Done, not just return.